app/controllers/auth_controller.rb

```diff
@@ -26,10 +29,20 @@ def token
         token_type: 'Bearer',
         expires_in: time_to_live,
         scope: scope,
-        patient: Patient.all.first.id
+        patient: Patient.first&.id
       }
+    elsif params.key?(:code) || params.key?(:client_id)
```

Comment:
I believe this is a public app workflow, so client id is required in the token exchange.

http://hl7.org/fhir/smart-app-launch/#step-3-app-exchanges-authorization-code-for-access-token

Reply:
In that case should we also check the client_id before handing the token? I.e:
```ruby
if params[:code] == <auth_code> && params[:client_id] == <client id>
    render json: <token stuff>
```

Reply:
^^ done